clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -32,6 +118,10 @@
         this.rate = rate;
     }
 
+    public abstract DataOutputStream wrapForOutput(ByteBufferOutputStream buffer, byte messageVersion, int bufferSize);
+
+    public abstract DataInputStream wrapForInput(ByteBufferInputStream buffer, byte messageVersion);
```

Comment:
Hmm.. That's fair. The annoying thing is that `ByteBufferInputStream` already implements `DataInputStream`, but maybe we could revert the change which provides that.